didcomm_messaging/quickstart.py

```diff
@@ -0,0 +1,393 @@
+        """Ensure that id is set."""
+        # super().__init__(*args, **kwargs)
+        if self.id is None:
+            self.id = str(uuid.uuid4())
+
+    def as_dict(self):
+        """Return message as a dictionary for serialization."""
+        return attrs.asdict(self, filter=(lambda _, x: x is not None))
+
+    @classmethod
+    def from_json(cls, data: str):
+        """Create a Message object from a JSON string."""
+        data = json.loads(data)
+        if "from" in data:
+            data["frm"] = data["from"]
+            del data["from"]
+        return cls(**data)
+
+
+class CompatibilityPrefixResolver(PrefixResolver):
+    """Provide backwards compatibility with older DID methods.
+
+    This will be removed in the future, as the intent is for agents to follow
+    the did:peer:2 spec. The CompatibilityPrefixResolver allows for interaction
+    with agents that are using the old #key-byte-prefix. Once this changes,
+    this class will be removed.
+
+    If you don't need this, it is recommended that you just use the
+    PrefixResolver directly.
+    """
+
+    async def resolve_and_parse(self, did: str) -> DIDDocument:
+        """Resolve a DID and parse the DID document."""
+        doc = await self.resolve(did)
+        # return DIDDocument.deserialize(doc)
+        id_map = {}
+
+        def set_id(method):
+            new_id = method["publicKeyMultibase"][1:9]
+            id_map[method["id"]] = new_id
+            method["id"] = did + "#" + new_id
+            return method
+
+        doc["verificationMethod"] = [
+            set_id(method) for method in doc["verificationMethod"]
+        ]
+        doc["authentication"] = [
+            did + "#" + id_map.get(id) for id in doc["authentication"]
+        ]
+        doc["keyAgreement"] = [did + "#" + id_map.get(id) for id in doc["keyAgreement"]]
+        return DIDDocument.deserialize(doc)
+
+
+def generate_did():
+    """Use Askar to generate encryption/verification keys, then return a DID from both."""
+
+    verkey = Key.generate(KeyAlg.ED25519)
+    xkey = Key.generate(KeyAlg.X25519)
+    did = generate(
+        [
+            KeySpec.verification(
+                multibase.encode(
+                    multicodec.wrap("ed25519-pub", verkey.get_public_bytes()),
+                    "base58btc",
+                )
+            ),
+            KeySpec.key_agreement(
+                multibase.encode(
+                    multicodec.wrap("x25519-pub", xkey.get_public_bytes()), "base58btc"
+                )
+            ),
+        ],
+        [
+            {
+                "type": "DIDCommMessaging",
+                "serviceEndpoint": {
+                    "uri": "didcomm:transport/queue",
+                    "accept": ["didcomm/v2"],
+                    "routingKeys": [],
+                },
+            }
+        ],
+    )
+    return did, (verkey, xkey)
+
+
+async def setup_default(did, did_secrets, enable_compatibility_prefix=False):
```

Comment:
I think this really needs type hinting. I highly value thorough and accurate type hints in code like this since it helps me (and my editor) to reason about the code.